### PR TITLE
Replacing source gen IndentedTextWriter with raw string literals

### PR DESCRIPTION
**Pull request by fabiocav**
Change to improve readability of source gen code.

### Pull request checklist

* [x] My changes **do not** require documentation changes
  * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes **should not** be added to the release notes for the next release
  * [ ] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
  * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] I have added all required tests (Unit tests, E2E tests)

### CODE REVIEW

**Comment by fabiocav** [comment]
@satvu I'll wait for your changes to land and rebase on them but wanted to share this.

**Review by kshyju — APPROVED** [review]

**Comment by kshyju** [comment]
Looks like the tests did not run for this PR build?

File: sdk/Sdk.Generators/FunctionMetadataProviderGenerator.Emitter.cs [hunk 1]
```diff
@@ -101,64 +77,44 @@ private void AddFunctionMetadataInfo(IndentedTextWriter indentedTextWriter, IRea
...
+            private string AddBindingInfo(string functionVarName, string functionBindingsListVarName, IList<IDictionary<string, object>> bindings)
             {
                 var bindingCount = 0;
-
+                var builder = new StringBuilder();
                 foreach (var binding in bindings)
                 {
                     var bindingVarName = functionVarName + "binding" + bindingCount.ToString();
-
                     var jsonBinding = JsonSerializer.Serialize(binding, _jsonOptions);
```
Comment (kshyju) [inline sdk/Sdk.Generators/FunctionMetadataProviderGenerator.Emitter.cs:110]: Unused.
`bindingCount` can also be removed after removing `bindingVarName`

File: sdk/Sdk.Generators/FunctionMetadataProviderGenerator.Emitter.cs [hunk 2]
```diff
@@ -1,98 +1,74 @@
...
+
+                return $$"""
+                         // <auto-generated/>
+                         using System;
+                         using System.Collections.Generic;
+                         using System.Collections.Immutable;
+                         using System.Text.Json;
+                         using System.Threading.Tasks;
+                         using Microsoft.Azure.Functions.Worker.Core.FunctionMetadata;
+                         using Microsoft.Azure.Functions.Worker.Http;
```
Comment (satvu) [inline sdk/Sdk.Generators/FunctionMetadataProviderGenerator.Emitter.cs:33]: Did you end up having to keep this import?

Comment (fabiocav) [reply sdk/Sdk.Generators/FunctionMetadataProviderGenerator.Emitter.cs:33]: I need to make sure we have good validation. Will keep it as part of these changes and send up update for the removal.

**Pull request merged**